CMakeLists.txt

```diff
@@ -213,6 +213,9 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "(GNU|Clang)")
     endif ()
 endif()
 
+# helpful for debugging installed executables, see `man ld-linux.so`
+set(CMAKE_INSTALL_RPATH "$ORIGIN/../lib")
```

Comment:
Do we really want this? I've had bad experiences with rpaths.

Reply:
It should be for comfort only, I'll try without.

Reply:
Specifically, this is the difference:

With:

```
$ DESTDIR=/tmp/test ninja install
$ /tmp/test/usr/local/bin/aktualizr --version
# works
```

without:

```
$ DESTDIR=/tmp/test ninja install
$ /tmp/test/usr/local/bin/aktualizr --version
# does not work
$ LD_LIBRARY_PATH=/tmp/test/usr/local/bin /tmp/test/usr/local/bin/aktualizr --version
# works
```

But then, it might be bad practice and cause other issues. Do you remember some details about the issues you had?

Reply:
The problem with using rpath is that you're forcing your path structure on your users. It means that when dynamically linking, everything in `"$ORIGIN/../lib"` will be taken over anything in the user's `$LD_LIBRARY_PATH`, which can have unintended side effects.

There are two points of view on this depending on what you think one's users prefer. One is that rpath makes it easier to find your library, but the downside is messing around with link path searching. The other is that rpath breaks traditional Linux norms and you should only really install things into default locations or change your `PATH` and/or `LD_LIBRARY_PATH` accordingly.

My preference is the latter, but I'm not set on that. Our circumstances are quite different than my past experience. If this makes things easier, it's fine. It does seem it'd make testing locally a bit easier.

Reply:
My view is that if a user installs it to a non-standard path, it is up to them to adjust LD_LIBRARY_PATH accordingly.

Reply:
> The problem with using rpath is that you're forcing your path structure on your users. It means that when dynamically linking, everything in `"$ORIGIN/../lib"` will be taken over anything in the user's `$LD_LIBRARY_PATH`, which can have unintended side effects.

According to the ld-linux.so's manual page, and a quick test, this doesn't seem to be the case. What CMake calls `RPATH` is actually `RUNPATH` which has a lower priority than `LD_LIBRARY_PATH`:

```
readelf -a usr/local/bin/aktualizr | grep PATH
 0x000000000000001d (RUNPATH)            Library runpath: [$ORIGIN/../lib]
```

That said, the risks of confusion might indeed outweigh the benefits, so I've removed it for now.

Reply:
> What CMake calls `RPATH` is actually `RUNPATH`

Wow, that is actually the most confusing part of all of this! If it's actually just setting `RUNPATH`, I don't really mind it. For now, though, if you've already removed it, let's see how it goes without it, and we can add it later if we think it'd be helpful.